src/utils/general.ts

```diff
@@ -122,76 +125,94 @@ function evaluate(
 
 const aggregateMap: {
```

Comment:
`[aggregateMap]` It would be nice to add some tests using this utility with different values to validates the new behavior.